src/main/java/baritone/utils/BaritoneProcessHelper.java

```diff
@@ -37,4 +61,105 @@ public BaritoneProcessHelper(Baritone baritone) {
     public boolean isTemporary() {
         return false;
     }
+
+    public void returnhome() {
+        IWaypoint waypoint = baritone.getWorldProvider().getCurrentWorld().getWaypoints().getMostRecentByTag(IWaypoint.Tag.HOME);
+        if (waypoint != null) {
+            Goal goal = new GoalBlock(waypoint.getLocation());
+            baritone.getCustomGoalProcess().setGoalAndPath(goal);
+        } else {
+            logDirect("No recent waypoint found, can't return home");
+        }
+    }
+
+    public boolean putInventoryInChest(Set<Item> validDrops) {
+        List<Slot> chestInv = ctx.player().openContainer.inventorySlots;
+        NonNullList<ItemStack> inv = ctx.player().inventory.mainInventory;
+
+        for (int i = 0; i < inv.size(); i++) {
+            if (!inv.isEmpty() && validDrops.contains(inv.get(i).getItem())) {
+                for (int j = 0; j < chestInv.size() - inv.size(); j++) {
+                    if (chestInv.get(j).getStack().isEmpty()) {
+                        ctx.playerController().windowClick(ctx.player().openContainer.windowId, i < 9 ? chestInv.size() - 9 + i : chestInv.size() - inv.size() + i - 9, 0, ClickType.QUICK_MOVE, ctx.player());
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    public boolean isInventoryFull() {
+        NonNullList<ItemStack> inv = ctx.player().inventory.mainInventory;
+        boolean inventoryFull = true;
+
+        for (ItemStack stack : inv) {
+            if (stack.isEmpty()) {
+                inventoryFull = false;
+                break;  
+            }
+        }
+
+        return inventoryFull;
+    }
+
+    public Set<ItemStack> notFullStacks(Set<Item> validDrops) {
+        NonNullList<ItemStack> inv = ctx.player().inventory.mainInventory;
+
+        Set<ItemStack> stacks = inv.stream()
+            .filter(stack -> validDrops.contains(stack.getItem()) && stack.getMaxStackSize() != stack.getCount())
+            .collect(Collectors.toCollection(HashSet::new));
+        
+        return stacks;
+    }
+
+    public BlockOptionalMetaLookup getBlacklistBlocks(Set<ItemStack> notFullStacks, BlockOptionalMetaLookup filter) {
+        List<BlockOptionalMeta> blacklistBlocks = new ArrayList<>();
+
+        for (BlockOptionalMeta bom : filter.blocks()) {
+            boolean blacklist = true;
+            for (ItemStack stack : notFullStacks) {
+                if (bom.matches(stack)) {
+                    blacklist = false;
```

Comment:
Instead of this blacklist boolean flag, label the outer loop (the bom one) with like `outer:` and then replace this flag set and break with `continue outer;`